_posts/2017-12-07-mixer-spof-myth.md

```diff
@@ -0,0 +1,108 @@
+
+* **Statelessness**. Mixer is stateless in that it doesn’t manage any persistent storage of its own.
+
+* **Hardening**. Mixer proper is designed to be a highly reliable component. The design intent is to achieve > 99.999% uptime for any individual Mixer instance.
+
+* **Caching and Buffering**. Mixer is designed to accumulate a large amount of transient ephemeral state.
+
+The sidecar proxies that sit next to each service instance in the mesh must necessarily be frugal in terms of memory consumption, which constrains the possible amount of local caching and buffering. Mixer, however, lives independently and can use considerably larger caches and output buffers. Mixer thus acts as a highly-scaled and highly-available second-level cache for the sidecars.
+
+Mixer’s expected availability is considerably higher than most infrastructure backends (those often have availability of perhaps 99.9%). Its local caches and buffers help mask infrastructure backend failures by being able to continue operating even when a backend has become unresponsive.
+
+## Mixer: Latency slasher
+
+As we explained above, the Istio sidecars generally have fairly effective first-level caching. They can serve the majority of their traffic from cache. Mixer provides a much greater shared pool of second-level cache, which helps Mixer contribute to a lower average per-request latency.
+
+While it’s busy cutting down latency, Mixer is also inherently cutting down the number of calls your mesh makes to infrastructure backends. Depending on how you’re paying for these backends, this might end up saving you some cash by cutting down the effective QPS to the backends.
+
+## Work ahead
+
+We have opportunities ahead to continue improving the system in many ways.
+
+### Config canaries
+
+Mixer is highly scaled so it is generally resistant to individual instance failures. However, Mixer is still susceptible to cascading failures in the case when a poison configuration is deployed which causes all Mixer instances to crash basically at the same time (yeah, that would be a bad day). To prevent this from happening, config changes can be canaried to a small set of Mixer instances, and then more broadly rolled out. 
+
+Mixer doesn’t yet do canarying of config changes, but we expect this to come online as part of Istio’s ongoing work on reliable config distribution.
+
+### Cache tuning
+
+We have yet to fine-tune the sizes of the sidecar and Mixer caches. This is work will be focused on achieving the highest performance possible using the least amount of resources.
+
+### Cache sharing
+
+At the moment, each Mixer instance operates independently of all other instances. A request handled by one Mixer instance will not leverage data cached in a different instance. We will eventually experiment with a distributed cache such as memcached or Redis in order to provide a much larger mesh-wide shared cache, and further reduce the number of calls to infrastructure backends.
+
+### Sharding
```

Comment:
I don't understand this work item, sorry. May be I don't know the technical details or we are missing some explanation in this section.

Reply:
I rephrased. Hopefully it's more explicit now.